frontend/persist.lua

```diff
@@ -137,30 +96,26 @@ local codecs = {
     -- dump: human readable, pretty printed, fast enough for most use cases.
     dump = {
         id = "dump",
-        reads_from_file = true,
-        writes_to_file = false,
 
         serialize = function(t, as_bytecode)
-            local content
+            local ok, d = pcall(dump, t)
```

Comment:
This is the only place it's called directly (on account of the pcall that's necessary with these changes)?

Reply:
Sorry, I don't understand what you mean.

Reply:
Note that FWIR, the support for serializing / de-serializing functions with the serpent codec is clearly never used, so it could just be dropped.

Reply:
I mean that above you changed a line to `error("cannot dump functions")` which potentially makes the places where `dump()` is called dangerous, unless I'm missing something.

Reply:
> the support for serializing / de-serializing functions with the serpent codec is clearly never used

I kind of expect it to read `function: 0xetc`. Come to think about it, how about debug logging? It's quite useful there.

Reply:
This is the result with:
```lua
local table_with_function = { a = function() print("function") end, }
```
→ `do local _={a=((loadstring or load)("\27LJ\2\8!@spec/front/unit/persist_spec.lua.\0\0\3\0\2\0\4\5\5\0006\0\0\0'\2\1\0B\0\2\1K\0\1\0\13function\nprint\0\0\0\0\0\0",'@serialized'))};return _;end`

Reply:
I added an error trace, or do you really prefer debug level? There's a warning just below.

Reply:
Not dumping functions was very much on purpose to avoid garbage like that in logs, as `logger` uses `seperent` ;).

Reply:
(Well, not via persist, but it's basically just in there for funsies and I didn't want to introduce behavioral difference between logger.debug & persist + serpent).

Reply:
Haven't slept, sensible thoughts will have to wait for tomorrow; but the API clunk was also on purpose, to avoid temporary Lua objects when we could deal with native C objects directly without intermediaries.

Reply:
(i.e., looks like a NAK to me)

Reply:
As @NiLuJe said, I was talking about logger
debug. 

Just because it's not in the code doesn't mean we don't use it all the time. The behavior is intended to be fast and easy to read, but it hadn't clicked immediately that this would affect it.